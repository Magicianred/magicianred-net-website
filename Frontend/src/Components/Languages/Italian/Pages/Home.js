import React, { Component } from 'react';
import axios from 'axios';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Carousel from 'react-bootstrap/Carousel';

import Layout from './../Partials/Layout';
import Paragraph from '../../../../Models/Paragraph';
import HomeMessage from '../../../../Models/HomeMessage';

const publicUrl = process.env.PUBLIC_URL;

/**
 * Component for showing the Home page.
 * 
 * @component
 * @example
 * return (
 *   <Home />
 * )
 */
class Home extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            items: [],
            paragraphs: [],
            hasError: false,
        };
    }        

    componentDidMount() {
        const items = [];
        this.serverRequest = 
            axios
                .get(`${publicUrl}/data/languages/english/home_messages.json`)
                .then((result) => {
                    result.data.map((item, index) => items.push(new HomeMessage(index, item.title, item.text, item.imagesrc)));
                    this.setState({ items });
            });
        const paragraphs = [];
        this.serverRequest = 
            axios
                .get(`${publicUrl}/data/languages/english/pages/home.json`)
                .then((result) => {
                    result.data[0].paragraphs.map((item, index) => paragraphs.push(new Paragraph(index, item.text)));
                    this.setState({ paragraphs });
            });
    }

    // componentWillUnmount() {
    //     if (this.serverRequest) {
    //         this.serverRequest.abort();
    //     }
    // }

    static getDerivedStateFromError(error) {       
        return { hasError: true };  
    }    

    render() {
        const { items, paragraphs, hasError } = this.state;
        return (
            <Layout>
                {hasError ? <h1>Something went wrong.</h1> : null}
                <article>
							<h2><a href="#">Premessa</a></h2>
							<p>Al momento non ci sono articoli, ho solo impostato l'aspetto del sito ed alcune idee.</p>
						</article>
        		<article>
							<h2><a href="#">First Article Title</a></h2>
							<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi. Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Curabitur sodales ligula in libero. Sed dignissim lacinia nunc. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla.</p>
						</article>
						
        		<article>
							<h2><a href="#">Second Article Title</a></h2>
							<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi. Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Curabitur sodales ligula in libero. Sed dignissim lacinia nunc.</p>
							<p>Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.</p>
							<p>Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.</p>
						</article>
            </Layout>
        );
    }
}

export default Home;