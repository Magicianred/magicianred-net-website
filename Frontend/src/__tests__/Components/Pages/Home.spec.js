import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { create } from 'react-test-renderer';

import Home from "./../../../Components/Pages/Home";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("Show a home page", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<Home />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    it("Show home page", () => {
        act(() => {
            render(<Home />, container);
        });
        expect(container.querySelector("h2").textContent).toBe("Home page");
        expect(container.querySelector("p").textContent).toBe("This is the home page");
    });

    xit("Show the header", () => {
    });

    xit("Show the site menu", () => {
    });

    xit("Show the site footer", () => {
    });

    xit("Show first message to users", () => {
    });

    xit("I press the next button", () => {
    });
});