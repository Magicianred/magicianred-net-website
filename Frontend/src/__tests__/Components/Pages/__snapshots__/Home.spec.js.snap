// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show a home page Snapshots Matches the snapshot (with default title) 1`] = `
<div
  className="page"
>
  <div
    className="page home jumbotron"
  >
    <h2>
      Home page
    </h2>
    <p>
      This is the home page
    </p>
    <hr />
    <div
      className="carousel slide"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          onClick={[Function]}
        />
         
        <li
          className={null}
          onClick={[Function]}
        />
         
      </ol>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
        >
          <img
            alt="Welcome to new site!"
            className="d-block w-100"
            src="https://picsum.photos/200/100"
          />
          <div
            className="carousel-caption"
          >
            <h3>
              Welcome to new site!
            </h3>
            We are glade to present you our new web site
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Lorem ipsum dolor sit amet"
            className="d-block w-100"
            src="https://placeimg.com/200/100/tech/grayscale"
          />
          <div
            className="carousel-caption"
          >
            <h3>
              Lorem ipsum dolor sit amet
            </h3>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce nisl lacus, lobortis vel nulla ut, bibendum suscipit magna. Phasellus efficitur risus et ante sodales fringilla. Morbi massa dui, tempor id luctus ac, maximus a orci. In sit amet eros scelerisque magna dictum laoreet. Nunc ac mauris sed tellus mollis ornare. Curabitur aliquet diam in sapien luctus auctor.
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>
</div>
`;
